apps/src/music/views/SoundsPanel.tsx

```diff
@@ -22,21 +25,29 @@ interface FolderPanelRowProps {
   libraryGroupPath: string;
   playingPreview: string;
   folder: SoundFolder;
+  currentValue: SoundFolder;
+  onSelect: (path: SoundFolder) => void;
   onPreview: (path: string) => void;
+  currentFolderRefCallback: (ref: HTMLDivElement) => void;
 }
 
 const FolderPanelRow: React.FunctionComponent<FolderPanelRowProps> = ({
   libraryGroupPath,
   playingPreview,
   folder,
+  currentValue,
+  onSelect,
   onPreview,
+  currentFolderRefCallback,
 }) => {
   const previewSound = folder.sounds.find(sound => sound.preview);
-  const soundPath = previewSound && folder.path + '/' + previewSound.src;
+  const soundPath = previewSound && folder.id + '/' + previewSound.src;
```

Comment:
Just checking my understanding here - we still need to store these event IDs as `folder.id/sound.src` to maintain backwards compatibility, as old projects would have sounds in the format `beats/groovy_beat` and we don't want to break those, right? However then in the MusicPlayer, we replace the `folder.id/sound.src` path with a `folder.path/sound.src` path which becomes something like `paths/beats/groovy_beat`?

Once we've fully switched projects over, just thinking it would be good to unify these so we're always storing either `folder.path/sound.src` as the sample ID consistently, or maybe just storing the `sound.src` by itself

Reply:
> Just checking my understanding here - we still need to store these event IDs as `folder.id/sound.src` to maintain backwards compatibility, as old projects would have sounds in the format `beats/groovy_beat` and we don't want to break those, right? However then in the MusicPlayer, we replace the `folder.id/sound.src` path with a `folder.path/sound.src` path which becomes something like `paths/beats/groovy_beat`?

That is correct.  And with the latest work just merged, there is also an optional per-sound `path, so that it could even become `packs/pack1/beats/groovy_beat`.

> Once we've fully switched projects over, just thinking it would be good to unify these so we're always storing either `folder.path/sound.src` as the sample ID consistently, or maybe just storing the `sound.src` by itself

Yeah, I went back and forth quite a bit on the ideal form for the identifiers in the user's source code, and it would be great to talk through it some more.  We could do any of the following, with pros and cons for each:

- `groovy_beat`
- `pack1/groovy_beat` (existing)
- `packs/pack1/groovy_beat`
- `packs/pack1/beats/groovy_beat`